index.js

```diff
@@ -26,6 +26,8 @@ function mergeObject(target, source, options) {
 	Object.keys(source).forEach(function(key) {
 		if (!options.isMergeableObject(source[key]) || !target[key]) {
 			destination[key] = cloneUnlessOtherwiseSpecified(source[key], options)
+		} else if (options.customMerge) {
+			destination[key] = options.customMerge(key, options)(target[key], source[key], options)
```

Comment:
If the `customMerge` function doesn't return a function for the given key, we need to fall back  to the default deepmerge function – so we need two things:

1. we need a test that passes in a `customMerge` that returns a non-function result for some keys
2. somewhere in the global scope we could have
```js
function getMergeFunction(key, options) {
	if (!options.customMerge) {
		return deepmerge
	}
	var customMerge = options.customMerge(key)
	return typeof customMerge === 'function' ? customMerge : deepmerge
}
```
and then here in the `forEach` we can do
```js
destination[key] = getMergeFunction(key, options)(target[key], source[key], options)
```
without needing the `if (options.customMerge)` check at all.

Does that make sense?  I might need some more coffee.

Reply:
That looks great, I've commited these changes up already and added tests to cover situations where a function isn't returned from `customMerge`, they seem to cover the code fairly well but if that needs any more then let me know